src/vma/sock/sock-redirect.cpp

```diff
@@ -677,6 +677,37 @@ int vma_add_ring_profile(vma_ring_type_attr *profile, vma_ring_profile_key *res)
 	return 0;
 }
 
+extern "C"
+int vma_modify_ring(struct vma_modify_ring_attr *mr_data)
+{
+	srdr_logdbg_entry("ring_fd=%d, mr_data=%p ", mr_data->ring_fd, mr_data);
+	int ret = -1;
+	cq_channel_info* p_cq_ch_info = g_p_fd_collection->get_cq_channel_fd(mr_data->ring_fd);
+	if (likely(p_cq_ch_info)) {
+		ring_simple* p_ring = dynamic_cast<ring_simple*>(p_cq_ch_info->get_ring());
+		if (likely(p_ring)) {
+			if (VMA_MODIFY_RING_CQ_MODERATION & mr_data->comp_bit_mask) {
+				p_ring->modify_cq_moderation(mr_data->cq_moderation.cq_moderation_period_usec,
+						mr_data->cq_moderation.cq_moderation_count);
+				ret = 0;
+			} else if (VMA_MODIFY_RING_CQ_ARM & mr_data->comp_bit_mask) {
+				return p_ring->request_notification(CQT_RX_TX, 0);
+			} else {
+				vlog_printf(VLOG_ERROR, "comp_mask [0x%x] is not supported\n",
```

Comment:
Please use srdr_logerr macro - no need to add "\n" with this macro

Reply:
:-1: